v-next/hardhat-errors/src/errors.ts

```diff
@@ -2,45 +2,92 @@ import { CustomError } from "@nomicfoundation/hardhat-utils/error";
 
 import { ERRORS, ErrorDescriptor } from "./descriptors.js";
 
-export const ERROR_PREFIX = "HHE";
-
-const IS_HARDHAT_ERROR_PROPERTY_NAME = "_isHardhatError";
-
 export type ErrorMessageTemplateValue =
   | string
   | number
   | boolean
   | bigint
   | undefined
   | null
+  | ErrorMessageTemplateValue[]
```

Comment:
I am not following this move. Is this simply that in an error message you can include an array so:

```ts
 const error = new HardhatError(
        {
          number: 12,
          messageTemplate: "{a} {b} {c}",
          websiteTitle: "Title",
          websiteDescription: "Description",
        } as const,
        { a: [1,2,3] },
        new Error(),
      );
```

Was this not supported before and it is now, or did something else prompt the move?

Reply:
it was, because arrays have a `toString`, but I made it more explicit.